modulecmd.tcl.in

```diff
@@ -37,7 +37,14 @@ set reportfd stderr ;# File descriptor to use to report messages
 set g_pager {@pager@} ;# Default command to page into, empty=disable
 set g_pager_opts {@pageropts@} ;# Options to pass to the pager command
 
-set g_siteconfig @etcdir@/siteconfig.tcl ;# Site configuration
+proc getSiteConfig {} {
+   if {[info exists env(MODULES_SITECONFIG)]} {
+      return $env(MODULES_SITECONFIG)
+   }
+   return @etcdir@/siteconfig.tcl
+}
+
+set g_siteconfig [getSiteConfig] ;# Site configuration
```

Comment:
I would prefer that `getSiteConfig` follow the same scheme than `getAutoHandling`, which means it should set the ::g_siteconfig if it is not set and returns its value in any cases. To set the global variable, it uses the `@etcdir@/siteconfig.tcl`value by default or the env(MODULES_SITECONFIG) if set.

Reply:
By this, do you also mean that any pieces of downstream code reference `[getSiteConfig]` rather than referring to `$g_siteconfig` ?

for example this:
```tcl
   if {[file readable $g_siteconfig]} {
      reportDebug "Source site configuration ($g_siteconfig)"
      if {[catch {source $g_siteconfig} errMsg]} {
```
to this:
```tcl
   if {[file readable [getSiteConfig]]} {
      reportDebug "Source site configuration ([getSiteConfig])"
      if {[catch {source [getSiteConfig]} errMsg]} {
```

Reply:
Yes, exactly.